### PR TITLE
alias __setattr__ to __setitem__ in Results

### LINKED ISSUE #3282: Current Results class has a conflict in key validation between setattr and setitem
## Expected behavior ##

I can modify the *same value* in a Results class both by using `__setitem__` and `__setattr__`. e.g.

```python
def test_setattr_modify_item(self, results):
        setattr(results, "myattr", 0)
        assert results.myattr == 0
        results["myattr"] = 3
        assert results.myattr == 3
```

## Actual behavior ##

The [above test fails](https://github.com/MDAnalysis/mdanalysis/pull/3281/commits/cb2c34edfebbc079ac530515474fe51dfd3fa44e). This only happens when actions follow a specific order:

- modify a value with setattr
- then modify that value with setitem
- then try to get back the modified value with getattr/ibute

This is because in the current Results code, `__setattr__` actually sets the attribute. When this happens, attribute access happens with `__getattribute__` rather than `__getattr__`. This means that while `__setitem__` is modifying the key/value pair in `Results.data`, there is also an attribute that *does not* get modified. This attribute is what gets returned on attribute access.

While it may be easy to just tell users not to mix-and-match the setattr/setitem behaviour, it's apparently natural enough that it's causing the helix_analysis tests to fail (https://github.com/MDAnalysis/mdanalysis/pull/3279).

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import PSF, DCD,  GRO, PDB, TPR, XTC, TRR,  PRMncdf, NCDF

u = mda.Universe(PSF, DCD)

....

```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
- Which version of Python (`python -V`)?
- Which operating system?

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #3282

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/base.py [hunk 1]
```diff
@@ -84,8 +84,11 @@ class in `scikit-learn`_.
 
     .. versionadded:: 2.0.0
     """
+
+    data = None
+
     def _validate_key(self, key):
-        if key in dir(UserDict) or (key == "data" and self._dict_frozen):
+        if key in dir(self):
```
Comment (PicoCentauri) [inline package/MDAnalysis/analysis/base.py:91]: This is nice now!

File: package/MDAnalysis/analysis/base.py [hunk 2]
```diff
@@ -94,28 +97,25 @@ def _validate_key(self, key):
...
         if "data" in kwargs.keys():
             raise AttributeError(f"'data' is a protected dictionary attribute")
-
-        self._dict_frozen = False
-        for key in kwargs:
-            self._validate_key(key)
-        super().__init__(**kwargs)
-        self._dict_frozen = True
+        self.__dict__["data"] = {}
+        self.update(kwargs)
```
Comment (PicoCentauri) [inline package/MDAnalysis/analysis/base.py:101]: [`UserDict`](https://github.com/python/cpython/blob/7424105a02bbc209e032752636bae30ccb556402/Lib/collections/__init__.py#L1040-L1048) allow to initlize the instance with passing a dictionary. This does not work anymore. I suggest you copy the four lines 
```python
if dict is not None:
    self.update(dict)
if kwargs:
    self.update(kwargs)
```

from cpython, adjust the arguments, add a test and everything is fine.

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
This is a smart PR!

File: package/MDAnalysis/analysis/base.py (hunk 2 above)
Comment (lilyminium) [reply package/MDAnalysis/analysis/base.py:102]: Are you happier with [my somewhat lazier solution? :) ](https://github.com/MDAnalysis/mdanalysis/pull/3281/commits/619d9374851c4b3c8e45f3a95f0181b080288285)

**Review by orbeckst — APPROVED** [review]
Looks good to me.

(I was debating if we should have a test showing that two instances do not interfere with each other but that seems un-necessary with the current implementation — it just makes use of the way that Python resolves class and instance attribute access.)

**Comment by orbeckst** [comment]
@PicoCentauri are you happy with the changes?

@PicoCentauri @lilyminium please merge when ready.

File: package/MDAnalysis/analysis/base.py [hunk 3]
```diff
@@ -84,38 +84,39 @@ class in `scikit-learn`_.
 
     .. versionadded:: 2.0.0
     """
+
+    data = None
+
```
[the remark this answers is not in the document]
Comment (lilyminium) [reply package/MDAnalysis/analysis/base.py]: Woops, that's actually an unnecessary remnant of earlier code. Writing into `__dict__` is about the hackiest, most guaranteed way to make sure you're changing the object instance without calling any dunder methods.

**Comment by lilyminium** [comment]
Should I update CHANGELOG? The Results PR itself was very recent.

File: package/MDAnalysis/analysis/base.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (orbeckst) [reply package/MDAnalysis/analysis/base.py]: I thought you needed this for the check on `__init__` that no existing attrs were included. But given that you add `data` to `__dict__` before anything else happens, this is not needed.

**Comment by orbeckst** [comment]
I'd say no — normally we update CHANGELOG for anything that has an associated issue. But in this case this is a hotfix and the CHANGELOG entry would be more confusing than anything else.

**Review by cbouy — APPROVED** [review]
Sorry didn't get much time but all looks good!

File: package/MDAnalysis/analysis/base.py (hunk 2 above)
Comment (PicoCentauri) [reply package/MDAnalysis/analysis/base.py:100]: Totally!

**Pull request merged**